src/urllib3/_request_methods.py

```diff
@@ -208,9 +209,8 @@ def request_encode_body(
                 )
 
             extra_kw["body"] = body
-            extra_kw["headers"] = {"Content-Type": content_type}
+            extra_kw["headers"].add("Content-Type", content_type)
```

Comment:
I added an implementation of `setdefault` to the `HTTPHeaderDict`, though in order to keep the stuff a bit straightforward I made the default to `''` instead of `None` (including a default value was needed to match the `MutableMapping` signature for `setdefault`)